examples/slackbot/Notebook.ipynb

```diff
@@ -27,89 +27,93 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "id": "e95d4d50",
-   "metadata": {},
-   "outputs": [],
+   "metadata": {
+    "collapsed": true,
+    "jupyter": {
+     "outputs_hidden": true
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "ename": "ModuleNotFoundError",
+     "evalue": "No module named 'openai'",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
+      "\u001b[0;32m/tmp/ipykernel_546/897995286.py\u001b[0m in \u001b[0;36m<cell line: 3>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mdatetime\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mdatetime\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimedelta\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0msklearn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmodel_selection\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mtrain_test_split\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0;32mimport\u001b[0m \u001b[0mopenai\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      4\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mkaskada\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mk\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
+      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'openai'"
+     ]
+    }
+   ],
    "source": [
     "from datetime import datetime, timedelta\n",
+    "from sklearn.model_selection import train_test_split\n",
     "import openai\n",
     "import kaskada as k\n",
     "\n",
-    "# Download Slack history from their API\n",
-    "# -> https://slack.com/help/articles/201658943-Export-your-workspace-data\n",
-    "# -> https://api.slack.com/events/message \n",
-    "# Example Event:\n",
-    "# { \t\n",
-    "#   \"type\": \"message\", \t\n",
-    "#   \"channel\": \"C123ABC456\", \t\n",
-    "#   \"user\": \"U123ABC456\", \t\n",
-    "#   \"text\": \"Hello world\", \t\n",
-    "#   \"ts\": \"1355517523.000005\" \n",
-    "# }\n",
+    "def prompt(messages):\n",
```

Comment:
What is the type of messages here?

Reply:
Not sure - `Expr`? For clarity in the demo I'm not sure if it's worth using type annotations, unless it helps communicate what's happening.